RawEventRecord.cpp

```diff
@@ -38,20 +39,68 @@ class RecordFieldIterator {
             idx = _str.size();
         }
         _val = _str.substr(_idx, idx-_idx);
-        // For certain record types, the data is inside a "msg='...'" field.
-        if (_val.substr(0, 5) == SV_MSG) {
-            _idx+=5;
-            return next();
-        } else {
-            _idx = _str.find_first_not_of(SV_WSP, idx);
+        _idx = _str.find_first_not_of(SV_WSP, idx);
+        return true;
+    }
+
+    // Advance to the next key=value
+    bool next_kv() {
+        static auto SV_MSG = "msg"sv;
+        static auto SV_WSP = " \n"sv;
+        static auto SV_WSPQ = "' \n"sv;
+        static auto SV_SP = " "sv;
+        static auto SV_EQ = "="sv;
+        static auto SV_DQ = "\""sv;
+
+        if (_idx == std::string_view::npos || _idx >= _str.size()) {
+            return false;
         }
-        // The field might have been inside a "msg='...'" so ignore the "'"
-        if (_val.back() == '\'') {
-            _val = _val.substr(0, _val.size()-1);
+
+        // Find the '='
+        auto idx = _str.find_first_of(SV_EQ, _idx);
+        if (idx == std::string_view::npos) {
+            // No '=' found, assume remainder of text is unparsable
+            idx = _str.size();
+            _val = _str.substr(_idx, idx-_idx);
+            _key = std::string_view(); // Make _key empty to signal that _val has remainder of text
+            return true;
+        }
+
+        _key = _str.substr(_idx, idx-_idx);
+        idx += 1; // Skip past the '='
+        _idx = idx; // Set _idx to start of value
+
+        // For certain record types, some of the data is inside a "msg='...'" field.
+        if (_key == SV_MSG && _str[_idx] == '\'') {
+            _idx += 1; // Skip past the (') char
+            return next_kv();
+        } else {
+            if (_str[_idx] == '"') {
+                // Value is double quoted, look for end quote
```

Comment:
Could double quotes be escaped?

Reply:
No, if the value contains a double quote, then entire value is HEX encoded.